tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -350,84 +357,92 @@ func run(config *config.Config) {
+		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-o", "lo", "!", "-d", "127.0.0.1/32", "-j", constants.ISTIOINREDIRECT)
 	}
 
-	for _, uid := range split(config.ProxyUID) {
+	for _, uid := range split(iptConfigurator.cfg.ProxyUID) {
 		// Avoid infinite loops. Don't redirect Envoy traffic directly back to
 		// Envoy for non-loopback traffic.
-		ext.RunOrFail(dep.IPTABLES, "-t", constants.NAT, "-A", constants.ISTIOOUTPUT, "-m", "owner", "--uid-owner", uid, "-j", constants.RETURN)
+		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-m", "owner", "--uid-owner", uid, "-j", constants.RETURN)
 	}
 
-	for _, gid := range split(config.ProxyGID) {
+	for _, gid := range split(iptConfigurator.cfg.ProxyGID) {
 		// Avoid infinite loops. Don't redirect Envoy traffic directly back to
 		// Envoy for non-loopback traffic.
-		ext.RunOrFail(dep.IPTABLES, "-t", constants.NAT, "-A", constants.ISTIOOUTPUT, "-m", "owner", "--gid-owner", gid, "-j", constants.RETURN)
+		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-m", "owner", "--gid-owner", gid, "-j", constants.RETURN)
 	}
 	// Skip redirection for Envoy-aware applications and
 	// container-to-container traffic both of which explicitly use
 	// localhost.
-	ext.RunOrFail(dep.IPTABLES, "-t", constants.NAT, "-A", constants.ISTIOOUTPUT, "-d", "127.0.0.1/32", "-j", constants.RETURN)
+	iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-d", "127.0.0.1/32", "-j", constants.RETURN)
 	// Apply outbound IPv4 exclusions. Must be applied before inclusions.
 	for _, cidr := range ipv4RangesExclude.IPNets {
-		ext.RunOrFail(dep.IPTABLES, "-t", constants.NAT, "-A", constants.ISTIOOUTPUT, "-d", cidr.String(), "-j", constants.RETURN)
+		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-d", cidr.String(), "-j", constants.RETURN)
+	}
+
+	for _, internalInterface := range split(iptConfigurator.cfg.KubevirtInterfaces) {
+		iptConfigurator.iptables.InsertRuleV4(constants.PREROUTING, constants.NAT, 1, "-i", internalInterface, "-j", constants.RETURN)
 	}
 
-	for _, internalInterface := range split(config.KubevirtInterfaces) {
-		ext.RunOrFail(dep.IPTABLES, "-t", constants.NAT, "-I", constants.PREROUTING, "1", "-i", internalInterface, "-j", constants.RETURN)
+	iptConfigurator.handleInboundIpv4Rules(ipv4RangesInclude)
+	iptConfigurator.handleInboundIpv6Rules(ipv6RangesExclude, ipv6RangesInclude)
+
+	// Execute iptables commands
+	for _, cmd := range iptConfigurator.iptables.BuildV4() {
+		parts := strings.Split(cmd, " ")
+		iptConfigurator.ext.RunOrFail(parts[0], parts[1:]...)
 	}
 
-	handleInboundIpv4Rules(ext, config, ipv4RangesInclude)
-	handleInboundIpv6Rules(ext, config, ipv6RangesExclude, ipv6RangesInclude)
+	for _, cmd := range iptConfigurator.iptables.BuildV6() {
```

Comment:
I think we should do this only if we've detected that IPv6 is enabled.

Reply:
no.. check https://github.com/istio/istio/pull/17914/files#diff-7737d52cc5c9db7a81451826f15ec97eR183